compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -3542,7 +3542,7 @@ TR::Register *addConstantToLong(TR::Node *node, TR::Register *srcReg,
       {
       generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, trgReg, srcReg, value >> 16);
       if (value & 0x7fff)
-         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi2, node, trgReg, trgReg, value & 0x7fff);
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi2, node, trgReg, trgReg, LO_VALUE(value));
```

Comment:
I am not sure this is equivalent to the old behaviour. As I understand, before we were taking a 16-bit pattern and placed it into the instruction. Now we expect those 16-bit to be sign-extended to 32-bit before passing to generateTrg1Src1Instruction.

So places like value & 0xffff need to be fixed. But here we have (value & 0x7fff) so there is no sign-extension issue and potentially different value can be passed since we did not clear upper bit in the 16-bit value.

Reply:
I can change this back. I think I originally changed this because I used to call `HI_VALUE` to get the higher bits before realizing it wasn't necessary.